cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -696,7 +706,9 @@ Future<KafkaReconciler> kafkaReconciler()   {
                             currentPods.put(sts.getMetadata().getName(), IntStream.range(0, sts.getSpec().getReplicas()).mapToObj(i -> KafkaResources.kafkaPodName(kafkaAssembly.getMetadata().getName(), i)).toList());
                         }
 
-                        return new KafkaClusterCreator(vertx, reconciliation, config, kafkaMetadataStateManager.getMetadataConfigurationState(), supplier)
+                        KafkaClusterCreator kafkaClusterCreator =
+                                new KafkaClusterCreator(vertx, reconciliation, config, kafkaMetadataStateManager.getMetadataConfigurationState(), supplier);
+                        return kafkaClusterCreator
```

Comment:
Why this change?

Reply:
Because I need the `kafkaClusterCreator` instance withing the compose to access ...

```java
scalingDownBlockedNodes = kafkaClusterCreator.scalingDownBlockedNodes();
```

Unless we want to take the advice by @katheris here https://github.com/strimzi/strimzi-kafka-operator/pull/10583#discussion_r1763251535 and trying to put the `scalingDownBlockedNodes` down into the `KafkaCluster` instance and accessing it here in the lambda.

Reply:
No, I guess the explanation is sufficient for me.